mm/stopwordstripper.py

```diff
@@ -0,0 +1,114 @@
+#Imports
+import csv
+import os
+import sys
+import unicodedata
+
+#Main function.
+def main():
+    #load the tsv with unit tests.
+    with open(relpath(),'rb') as tsvin, open ('cases_tested.tsv','wb') as tsvout:
+        tsvin = csv.reader(tsvin,delimiter='\t')
+        tsvout = csv.writer(tsvout,delimiter='\t')
+
+        #Iterate through rows in cases.tsv.
+        for row in tsvin:
+            strippedline = stopwords(row[0],row[1])     
+            tsvout.writerow([row[0],row[1],strippedline])
+            #Use assert to check results.
+            assert strippedline == row[2]
+
+#Function to define path of cases.tsv.
+def relpath():
+    scriptdir = os.path.dirname(__file__) 
+    relpath = "path/to/test/cases.tsv" 
+    absfilepath = os.path.join(scriptdir, relpath)
+    return absfilepath
+
+#Function to test for non-english characters.
+def isenglish(stringin):
+    try:
+        stringin.encode(encoding='utf-8').decode('ascii')
+    except UnicodeDecodeError:
+        return False
+    else:
+        return True
+
+#Function to test stop words in input string.
+def stopwords(langcode,checkline):
+    #Load checkline into list.
+    checkline = checkline.split()
+    #Use NFC normalization to ensure proper encoding.
+    for item in checkline:
+        item = item.decode('utf8')
+        item = unicodedata.normalize('NFC', unicode(item))
+    #Define stopdict (for loading in stop word dictionaries).
+    stopdict = []
+    largestopdict = []
+    #Define stripped (checkline after going through selective stripping).
+    strippedlist = []
+    #Define englishlist final list.
+    englishlist = []
+    #Open the stop dictionary file based on langcode.
+    if (langcode == "en"):
+        with open("en.txt","r") as file:
```

Comment:
1) This means "en.txt" exists in the same directory ("mm") as this source code. I think we should use relative path **('../data/en.txt')** to the data directory so as to maintain one source of data,  that way when more languages or stop words are added in the future, the code will not break.

2) Another reason is we are duplicating data sources if we need to copy "en.txt" to "mm" directory. I think a good practice would be for us to have program data in one place for maintainability. Hence the relative path to the data directory.

Reply:
Thanks for the input, I will take a look at this.